### PR TITLE
e2e: serial: add test for related objs

### PR DESCRIPTION
**Pull request by ffromani**
Complement the current test we have for related objects with a e2e test which actually queries the cluster to crosscheck the related objects.

To enable the test, move the related objects code to an internal package. This code layour is however debatable and can be revisited in the future.

### CODE REVIEW

File: test/e2e/serial/tests/configuration.go [hunk 1]
```diff
@@ -685,9 +689,78 @@ var _ = Describe("[serial][disruptive][slow] numaresources configuration managem
...
+			err = fxt.Client.Get(context.TODO(), nroSchedKey, &nroSchedObj)
+			Expect(err).ToNot(HaveOccurred(), "cannot get %q in the cluster", nroSchedKey.String())
+
+			dps, err := objects.GetDeploymentOwnedBy(fxt.Client, nroSchedObj.ObjectMeta)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(dps)).To(Equal(1), "unexpected amount of scheduler deployments: %d", len(dps))
+
+			nrsExpected := objRefListToStringList(relatedobjects.Scheduler(dps[0].Namespace, nroSchedObj.Status.Deployment))
+			nrsGot := objRefListToStringList(nroSchedObj.Status.RelatedObjects)
+			Expect(nrsGot).To(Equal(nrsExpected), "mismatching related objects for NUMAResourcesScheduler")
```
Comment (shajmakh) [inline test/e2e/serial/tests/configuration.go:730]: ginkgo reporting can be handy for debugging and understanding what is being done. By() for the 2 main checks (one for NRO and one for the sched) is enough.

Comment (ffromani) [reply test/e2e/serial/tests/configuration.go:730]: right! let me add it

**Comment by shajmakh** [comment]
\lgtm

**Pull request merged**